cms/server/contest/handlers/taskusertest.py

```diff
@@ -125,12 +125,11 @@ class UserTestHandler(BaseHandler):
 
     @tornado.web.authenticated
     @actual_phase_required(0)
-    def post(self, task_name):
+    def post(self, contest_name, task_name):
         participation = self.current_user
 
         if not self.r_params["testing_enabled"]:
-            self.redirect("/")
-            return
+            raise tornado.web.HTTPError(403)
```

Comment:
This change is not needed for the overall patch, but I figured it was a sensible change (which didn't require a different commit or a different PR)